### PR TITLE
Sigma 1268 ussd baby DoB update

### PR DESCRIPTION
**Pull request by Buhle79**

### CODE REVIEW

File: test/ussd_popi_rapidpro.test.js [hunk 1]
```diff
@@ -414,6 +414,31 @@ describe("ussd_popi_rapidpro app", function() {
...
+                .setup.user.answer("contact", {fields: {preferred_channel: "WhatsApp",
+                    baby_dob1: "2021-03-10",
+                    baby_dob2: "2021-11-11",
+                    baby_dob3: "2022-07-07",
+                    edd: "2020-06-04T00:00:00Z",
+                    postbirth_messaging: "True",
+                },
+                })
+                .input("4")
+                .check.interaction({
```
Comment (erikh360) [inline test/ussd_popi_rapidpro.test.js:430]: Would be good to check that `answers.update_value` is set in this test

**Review by erikh360 — COMMENTED** [review]
One small comment

File: src/ussd_clinic_rapidpro.js [hunk 2]
```diff
@@ -1205,7 +1205,7 @@ go.app = function() {
             };
             var flow_uuid;
 
-            if (self.im.user.answers.state_message_type === "state_edd_month"
+            if (self.im.user.answers.state_message_type === "state_edd_year"
```
Comment (erikh360) [inline src/ussd_clinic_rapidpro.js]: Why this change?

**Review by erikh360 — COMMENTED** [review]
one more

File: src/ussd_popi_rapidpro.js [hunk 3]
```diff
@@ -700,21 +702,35 @@ go.app = function() {
...
+                }
+                else if (answers.update_value.startsWith("2nd")){
+                    data.change_baby = "baby_dob2";
+                }
+                else if (answers.update_value.startsWith("3rd")){
+                    data.change_baby = "baby_dob3";
+                }
+                data.change_type = "change_baby_dob";
+                data.baby_dob = baby_date,
+                end_flow = "state_baby_born_complete";
```
Comment (erikh360) [inline src/ussd_popi_rapidpro.js:729]: Is this still correct?

File: src/ussd_clinic_rapidpro.js (hunk 2 above)
Comment (Buhle79) [reply src/ussd_clinic_rapidpro.js]: state_message_type choice option are _state_edd_year_ for prebirth and _state_birth_year_ for postbirth. https://github.com/praekeltfoundation/ndoh-jsbox/blob/develop/src/ussd_clinic_rapidpro.js#L354

File: src/ussd_popi_rapidpro.js (hunk 3 above)
Comment (Buhle79) [reply src/ussd_popi_rapidpro.js:729]: Yes this is correct

**Comment by DevChima** [comment]
👍

File: src/ussd_clinic_rapidpro.js (hunk 2 above)
Comment (erikh360) [reply src/ussd_clinic_rapidpro.js]: But this is in the clinic line? The ticket is for baby dob change on the popi line?

Comment (Buhle79) [reply src/ussd_clinic_rapidpro.js]: Cool i will revert these changes
